frontend/src/components/ExcelUpload.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+import { ExcelToSchedule } from "../utils/excelParser"
+import { Schedule } from "../models/types"
+import styled from "styled-components";
+import IconButton from '@material-ui/core/IconButton';
+import Button from '@material-ui/core/Button';
+import Tooltip from '@material-ui/core/Tooltip';
+
+
+const Container = styled.div`
+    position: relative;
+    justify-content: right;
+    padding: 0px;
+    margin: 0px;
+`;
+
+
+interface props {
+    setSchedule: ((schedule: Schedule) => void)
+}
+
+export class ExcelUpload extends React.Component<props> {
+    constructor(props: any) {
+        super(props);
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    async handleChange(e : any) {
```

Comment:
Again this is super minor but I think a better name for this function would be `onUpload`